src/net/rtsp/request.cpp

```diff
@@ -136,9 +138,9 @@ static char* type_to_string(Request::Type type) {
 
 }
 
-bool Request::serialize(char* destination, std::size_t size) {
+ErrorOr<bool> Request::serialize(char* destination, std::size_t size) {
   if (!m_type.has_value() || !m_address.has_value() || !m_version.has_value()) {
-    throw std::runtime_error("One of fields is not initialized");
+    FAIL(Error::unitializedField);
```

Comment:
This is error of programmer, not user. `assert` should be used for these kinds of errors.